proxy/src/coco.rs

```diff
@@ -0,0 +1,467 @@
+//! Abstractions and utilities for git interactions through the API.
+
+use std::collections::hash_map::DefaultHasher;
+use std::hash::{Hash, Hasher};
+use std::str::FromStr;
+
+use librad::git;
+use librad::keys;
+use librad::meta::{self, common::Url};
+use librad::paths::Paths;
+use librad::peer;
+use librad::project;
+use radicle_surf as surf;
+use radicle_surf::git::git2;
+
+use crate::error;
+
+/// Branch name representation.
+#[derive(Debug, Eq, Ord, PartialEq, PartialOrd)]
+pub struct Branch(pub String);
+
+/// Tag name representation.
+///
+/// We still need full tag support.
+#[derive(Debug, Eq, Ord, PartialEq, PartialOrd)]
+pub struct Tag(pub String);
+
+/// Representation of a person (e.g. committer, author, signer) from a repository. Usually
+/// extracted from a signature.
+pub struct Person {
+    /// Name part of the commit signature commit.
+    pub name: String,
+    /// Email part of the commit signature commit.
+    pub email: String,
+    /// Reference (url/uri) to a persons avatar image.
+    pub avatar: String,
+}
+
+/// Representation of a code commit.
+pub struct Commit {
+    /// Identifier of the commit in the form of a sha1 hash. Often referred to as oid or object
+    /// id.
+    pub sha1: String,
```

Comment:
Any reason why we don't use a binary representation here?

Reply:
Leftover from the time these where directly used for the GraphQL API.